VisualizationBase/src/ViewItemManager.cpp

```diff
@@ -121,6 +124,57 @@ void ViewItemManager::removeAllViewItems()
 	currentViewItem_ = nullptr;
 }
 
+ViewItem* ViewItemManager::createIfNotExists(const QString name, QPoint position)
+{
+	//Try seeing if it is opened already
+	if (auto view = viewItem(name))
+		return view;
+	//Else check if we have it persisted somewhere
+	for (auto manager : Model::AllTreeManagers::instance().loadedManagers())
+		if (auto view = loadView(name, manager))
+		{
+			addViewItem(view, position);
+			return view;
+		}
+	//If it doesn't exist, create a new view item
+	return newViewItem(name, position);
+}
+
+void ViewItemManager::saveView(ViewItem* view, Model::TreeManager* manager) const
+{
+	auto json = view->toJson().toJson();
+	QFile file(fileName(view->name(), manager->name()));
+	file.open(QIODevice::WriteOnly);
+	QTextStream write(&file);
+	write << json;
+	file.close();
+}
+
+ViewItem* ViewItemManager::loadView(QString name, Model::TreeManager* manager)
+{
+	QFile file(fileName(name, manager->name()));
+	if (!file.exists())
+		return nullptr;
+	file.open(QIODevice::ReadOnly);
+	QTextStream read(&file);
+	QString json;
+	while (!read.atEnd())
+		json = json + read.readLine();
+	ViewItem* view = new ViewItem(nullptr);
+	view->fromJson(QJsonDocument::fromJson(json.toUtf8()));
+	return view;
+}
+
+QList<ViewItem*> ViewItemManager::viewItemsAsList() const
```

Comment:
I don't think you should have this method. I will comment more in the place you use it below.